File: csclub/2011/dec9/unscientific/sol.py
while True:
	n=input()
	if n<0:break;
	zeroes=0
	last=1
	while n>0:
		last=(last*(1,1,2,6,4)[n%5]*((6,2,4,8)[(n/5)%4]if n>4 else 1))%10
		n/=5
		zeroes+=n
	print str(last)+"u"+str(zeroes)

File: csclub/2011/dec9/unscientific/sol.c
#include<stdio.h>
int main(){
	const static int fact[]={1,1,2,6,4},pow2[]={6,2,4,8};
	for(;;){
		int n,zeroes=0,last;
		scanf("%d",&n);
		if(n<0)return 0;
		for(last=fact[n%5];n/=5;zeroes+=n)
			last=last*fact[n%5]*pow2[n%4]%10;
		printf("%du%d\n",last,zeroes);
	}
}
